api/src/main/java/org/eclipse/microprofile/faulttolerance/Retry.java

```diff
@@ -104,14 +104,24 @@
      *
      * @return Specify the failure to retry on. Only a type assignable from {@link java.lang.Error} or {@link java.lang.Exception} can be specified.
      * Specifying a custom java.lang.Throwable will be ignored.
+     * <p>
+     * Whenever the set of retry-on exceptions overlaps with the set of abort-on exceptions, {@code abortOn} takes precedence.
+     * Overlapping means not only that the same type is present in both sets, but also that one set contains a type which
+     * is a subtype of a type present in the other set. In practice, this means that whenever an exception is thrown,
+     * it is first matched against {@code abortOn}, and only if there's no match there, it is matched against {@code retryOn}.
      */
     @Nonbinding
     Class<? extends Throwable>[] retryOn() default { Exception.class };
 
     /**
-     * 
+     *
      * @return Specify the failure to abort on. Only a type assignable from {@link java.lang.Error} or {@link java.lang.Exception} can be specified.
      * Specifying a custom java.lang.Throwable will be ignored.
+     * <p>
+     * Whenever the set of retry-on exceptions overlaps with the set of abort-on exceptions, {@code abortOn} takes precedence.
+     * Overlapping means not only that the same type is present in both sets, but also that one set contains a type which
```

Comment:
I will rephrase to be:
If an exception is thrown, it first matches `{code abortOn}`. If the exception is listed in the `{code abortOn}` or can be assigned to one exception in the list, the exception will be aborted from the retry operation. 
If not, the exception will be matched against `{code retryOn}. If the exception is listed in the `{code retryOn}`  or can be assigned to one exception in the list, the exception will trigger the retry operation.

Reply:
I did a more significant rewording. Please take another look.